### DISCUSSION TITLE
Waveshare RP2040 Plus board

### DISCUSSION THREAD

**Discussion by RetiredWizard** [micropython/micropython#10309] [Core Development] [answered]
According to The Waveshare website, the RP2040 Plus is a software compatible board to the Raspberry Pi Pico that uses either a 4MB or 16MB flash chip, a USB-C connector and adds some battery recharge circuitry/connector. I have been using the standard downloaded Raspberry Pi Pico Micropython and Circuitpython images and other than only seeing 2MB of flash have not seen any issues.

Recently I took a look at building a Micropython image that uses the correct flash size and since it turns out that this board is supported in the pico-sdk folders all I had to do was create a folder named WAVESHARE_RP2040_PLUS_4MB in the ports/rp2/boards folder and populate it with an updated board.json, mpconfigboard.cmake and mpconfigboard.h files.

My question is, since these boards are so similar to a Raspberry Pi Pico, would it be worth submitting this board and the _16MB variant in a PR? And is that something that anyone can do or does it have to come from the vendor?

**Comment by peterhinch** [comment, marked as answer]
Anyone can do this.

**Comment by CroftAr** [comment]
> would it be worth submitting this board and the _16MB variant in a PR?

Please do  :-)  it's beyond me so far and the vendor supplied f/w is back at v1.17.

**Comment by mattytrentini** [comment]
Note that @RetiredWizard has indeed created PR #10314 to cover the Waveshare boards. 😄
